WorkflowUI/Sources/Screen/ScreenContaining.swift

```diff
@@ -0,0 +1,31 @@
+#if canImport(UIKit)
+
+import Foundation
+import UIKit
+
+/// Defines a type that semantically 'contains' a `Screen`, but erases its associated static type information.
+///
+/// This enables referring to `any ScreenViewController` which is not possible without a protocol
+/// like this because `ScreenViewController` is generic over its screen type and it's not possible to check
+/// `viewController is ScreenViewController` without also specifying a specific screen type.
+public protocol ScreenContaining {
```

Comment:
So I see this is still a draft, but how will this work for things like navigation controllers or tab bar controllers, where they have >= 1 inner screens?

Reply:
(I think you may want two properties, `containedScreens : [any Screen]` and `visibleScreens: [any Screen]`?)

Reply:
Those only have one current screen so it's ok for those to simply return the current "top" screen for conforming to this protocol.

The harder example is something like a split screen. In that case we may have to opt for tagging that screen with its own identity rather than trying to dig out the two individual screens its composing, but not sure.

Reply:
i think the main goals of this interface are to:

1. add parent type to ScreenViewController<T> so that its generic parameter can be ignored without losing access to its boxed Screen value
2. provide a mechanism for 'traversing' nested containers

i agree there are some questions about particular types of containers that might be worth further consideration, but i'm not sure if we need to deal with those questions at the moment or can punt them if this simple model works for most existing use cases.

Reply:
Maybe this spelling of the protocol should be `SingleScreenContaining`? That leaves the door open to a `ScreenContaining` that can handle multiple contained screens (where `SingleScreenContaining` could provide its own conformance to _that_ protocol).

I do also somewhat like calling this `visibleContainedScreen` (or similar) to make it clear that, for example, we’re interested in the topmost screen in a navigation stack not all the obscured screens.

Reply:
thanks for the feedback. i've made an update to rename to the protocol to `SingleScreenContaining`, and its requirement to `primaryScreen` to better communicate the intent. also had a go at improving the associated commentary.

i plan to work on the integration story with the intent to merge this early next week. if you have additional thoughts, do let me know!